File: Tema1/PrecisionMedicion.py
from math import pi

class PrecisionMedicion:
    @staticmethod
    def calcular(valor_original, valor_reportado):
        return abs(valor_original - valor_reportado)

    @staticmethod
    def area_circulo(r_original, r_reportado):
        area_real = pi * r_original ** 2
        area_reportada = pi * r_reportado ** 2
        return abs(area_real - area_reportada)


if __name__ == "__main__":
    p = PrecisionMedicion.calcular(3.45678, 3.457)
    print(f"Precisión de medición: {p}")

    a = PrecisionMedicion.area_circulo(5.678, 5.68)
    print(f"Precisión en área de círculo: {a}")

File: tema1/PrecisionMedicion.java
/**
 * Evalúa la precisión de una medición comparando dos valores.
 */
public class PrecisionMedicion {
    /**
     * Calcula la diferencia entre el valor original y el valor reportado.
     */
    public static double calcular(double valorOriginal, double valorReportado) {
        return Math.abs(valorOriginal - valorReportado);
    }

    /**
     * Calcula el error en el área de un círculo al usar un radio impreciso.
     */
    public static double areaCirculo(double rOriginal, double rReportado) {
        double areaReal = Math.PI * Math.pow(rOriginal, 2);
        double areaReportada = Math.PI * Math.pow(rReportado, 2);
        return Math.abs(areaReal - areaReportada);
    }

    public static void main(String[] args) {
        // Diferencia entre valor exacto y medido
        double precision = calcular(3.45678, 3.457);
        System.out.println("Precisión de medición: " + precision);

        // Error en el cálculo de un área
        double areaError = areaCirculo(5.678, 5.68);
        System.out.println("Precisión en área de círculo: " + areaError);
    }
}
